### PR TITLE
Admin add/edit menu

### PR DESCRIPTION
**Pull request by minza55113151**

### CODE REVIEW

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 1]
```diff
@@ -1,17 +1,173 @@
...
+          </Form.Item>
+
+          <Form.Item<FieldType>
+            name="price"
+            label="Price"
+            rules={[{ required: true, message: "Please input price!" }]}
+            style={{ width: "100%" }}
+          >
+            <Input placeholder="50" />
+          </Form.Item>
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:100]: Price should be the InputNumber.
Additional✨: with Baht icon and controls=false
[Antd Input ref](https://ant.design/components/input-number#input-number-demo-prefix) <- the first one of this example

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 2]
```diff
@@ -1,17 +1,173 @@
...
+            name="category"
+            label="Category"
+            style={{ width: "100%" }}
+          >
+            <Select placeholder="Breakfast" allowClear>
+              {categories.map((category) => (
+                <Select.Option value={category.name}>
+                  {category.name}
+                </Select.Option>
+              ))}
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:112]: Add key (@minza55113151  probably didn't install ESLINT vs code extension🤔)

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 3]
```diff
@@ -1,17 +1,173 @@
...
+            name="ingredients"
+            label="Ingredients"
+            style={{ width: "100%" }}
+          >
+            <Select mode="multiple" placeholder="Pork" allowClear>
+              {ingredientData.map((ingredient) => (
+                <Select.Option value={ingredient.name}>
+                  {ingredient.name}
+                </Select.Option>
+              ))}
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:126]: Add key

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 4]
```diff
@@ -1,17 +1,173 @@
...
+          >
+            <Input placeholder="50" />
+          </Form.Item>
+
+          <Form.Item<FieldType>
+            name="category"
+            label="Category"
+            style={{ width: "100%" }}
+          >
+            <Select placeholder="Breakfast" allowClear>
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:107]: Remove Placeholder

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 5]
```diff
@@ -1,17 +1,173 @@
...
+            label="Ingredients"
+            style={{ width: "100%" }}
+          >
+            <Select mode="multiple" placeholder="Pork" allowClear>
+              {ingredientData.map((ingredient) => (
+                <Select.Option value={ingredient.name}>
+                  {ingredient.name}
+                </Select.Option>
+              ))}
+            </Select>
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:127]: Change to select with mode tags this should be able to add new ingredients.
[AntD Select](https://ant.design/components/select#select-demo-size) <- the last one of this example

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 6]
```diff
@@ -30,3 +186,40 @@ const MenuFormCard = styled(Card)`
...
+const GeneralFormItemsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 4px;
+  width: 50%;
+`;
+
+const ImageFormItemsContainer = styled.div`
+  display: "flex";
```
Comment (katisd) [inline src/modules/admin/menu/MenuFormSection.tsx:214]: ??? Why "flex"🥹.

File: src/modules/admin/mock/ingredient.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    id: "2",
+    name: "Chicken",
+  },
+  {
+    id: "3",
+    name: "Beef",
+  },
+  {
+    id: "4",
+
```
Comment (katisd) [inline src/modules/admin/mock/ingredient.ts:21]: Pls remove I beg you👏👏👏🤣.

**Review by katisd — CHANGES_REQUESTED** [review]
Wait for others to check CSS. 🛌💤💤

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 1 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:100]: Done.

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 2 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:112]: Thx for ESLINT.
Done.

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 3 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:126]: Done

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 4 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:107]: Done

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 6 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:214]: Done.

File: src/modules/admin/mock/ingredient.ts (hunk 7 above)
Comment (minza55113151) [reply src/modules/admin/mock/ingredient.ts:21]: Done.

File: src/modules/admin/menu/MenuFormSection.tsx (hunk 5 above)
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:127]: Done.

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 8]
```diff
@@ -1,17 +1,178 @@
...
+
+  if (menuLayoutVariant === "edit") {
+    // TODO: get menu data from api then set value to form
+  }
+
+  const onFormSubmit = (values: any) => {
+    // TODO: send data to api
+    console.log(values);
+  };
+
```
[the remark this answers is not in the document]
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:40]: Done

File: src/modules/admin/menu/MenuFormSection.tsx [hunk 9]
```diff
@@ -1,17 +1,178 @@
...
+            label="Price"
+            rules={[{ required: true, message: "Please input price!" }]}
+            style={{ width: "100%" }}
+          >
+            <InputNumber
+              prefix="฿"
+              controls={false}
+              placeholder="50"
+              style={{ width: "100%" }}
+            />
```
[the remark this answers is not in the document]
Comment (minza55113151) [reply src/modules/admin/menu/MenuFormSection.tsx:104]: Done.

**Review by GolfGrab — APPROVED** [review]
nice work pls merge develop branch in to this branch before closing pr

**Comment by minza55113151** [comment]
@katisd pls approve to end this PR

**Review by katisd — APPROVED** [review]
Nice!

**Pull request merged**